### PR TITLE
Add onSwatchHover to CompactPicker

### PR DESCRIPTION
**Pull request by pconerly**
@casesandberg 

For #365

I only added `onSwatchHover` for `CompactPicker`, but if you'd like me to add it to any picker that wraps Swatch, I'd be happy to. And in general, I'm open to any ideas for getting this functionality in.

### CODE REVIEW

**Comment by casesandberg** [comment]
This looks stellar! Would you mind adding it for all of the pickers?

File: package.json [hunk 1]
```diff
@@ -22,7 +22,8 @@
     "clean": "rm -rf lib && mkdir lib",
     "lib": "npm run clean && babel src -d lib",
     "docs": "node ./scripts/docs-server",
```
Comment (casesandberg) [inline package.json]: Nit-picky, can you change the base "docs" command to `npm run docs-server` so that we don't have duplicate command code in there?

Comment (pconerly) [inline package.json]: Done!

**Comment by pconerly** [comment]
@casesandberg so I added it to all the relevant higher-level components (`Block`, `Circle`, `Compact`, `Github`, `Sketch`, `Swatches`, `Twitter`)

I also added tests to test the eventing. I struggled doing it with the existing test utils, so I added airbnb's `enzyme`. If that's a no-go LMK and I can try to figure out another way, but I think enzyme is really great.


I also found that I had unnecessary code in `Compact` where I was making the `optionalEvents` object that was splatted into it.  Only `Swatch` needs to do the null check, all of the other components can happily pass `null`s around.

File: package.json [hunk 2]
```diff
@@ -17,11 +17,12 @@
     "posttest": "npm run eslint",
     "prepublish": "npm run lib",
     "jest": "node_modules/.bin/jest",
+    "jest-watch": "node_modules/.bin/jest --watch",
     "eslint": "node_modules/.bin/eslint src/**/*.js",
     "test": "npm run unit-test && npm run jest",
     "clean": "rm -rf lib && mkdir lib",
     "lib": "npm run clean && babel src -d lib",
-    "docs": "node ./scripts/docs-server",
+    "docs-server": "node ./scripts/docs-server",
```
Comment (casesandberg) [inline package.json]: Ah, sorry, should have been more clear here. Please leave both `docs` and `docs-server` but have the docs script just call the docs-server one: 
```
"docs": "npm run docs-server",
```

**Comment by casesandberg** [comment]
This is wonderful! It has been on my todo to add more comprehensive tests with enzyme, so this is perfect! Just one last thing with the run scripts and I can merge this in!

**Comment by pconerly** [comment]
Great to hear :) I'll fix the package.json thing.

I'm also not sure if eslint is working properly? I'm not getting any errors, but the `eslintrc.js` references ` extends: '@case/eslint-config' ` ? And I'm not sure how that works.

**Comment by casesandberg** [comment]
Eslint is working-- You wrote the code with the right formatting, so there are no errors!

**Comment by casesandberg** [comment]
This is live in `react-color@2.13.0` 🎉  Thanks for such a great first contribution!

**Comment by pconerly** [comment]
Thanks for your help too :) It was easy to add this change in such a well-organized project!

**Pull request merged**